warden/src/server.rs

```diff
@@ -13,20 +19,16 @@ use tokio_stream::{
 use tonic::{Request, Response, Status};
 use tracing::{debug, info, instrument};
 
+use crate::assignment_computation::{AssignmentComputation, SimpleAssignmentComputation};
+
 /// Implementation of the Warden service.
-///
-/// It receives new assignment versions via a broadcast channel and uses a callback function
-/// to get the full or delta assignment updates. It sends the updates to the clients via the
-/// `register_range_server` streaming gRPC method.
-pub struct WardenServer {
-    // TODO(purujit): Replace these two fields with a MapProducer trait that exposes these members.
-    pub update_sender: broadcast::Sender<i64>,
-    update_callback: fn(i64, bool) -> Option<WardenUpdate>,
+pub struct WardenServer<'a> {
```

Comment:
why are the lifetime parameters needed here given you're using arc? is that an implication of the dyn?

Reply:
Good catch. It was left over from before when I didn't have the `Arc`. Removed now.